dataflow/src/main/java/org/checkerframework/dataflow/reachingdefinitions/ReachingDefinitionsStore.java

```diff
@@ -0,0 +1,121 @@
+package org.checkerframework.dataflow.reachingdefinitions;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.dataflow.analysis.Store;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.dataflow.cfg.visualize.CFGVisualizer;
+import org.checkerframework.dataflow.expression.JavaExpression;
+import org.checkerframework.javacutil.BugInCF;
+
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.StringJoiner;
+
+/** A reaching definitions store records information about ReachingDefinitionsNode */
+public class ReachingDefinitionsStore implements Store<ReachingDefinitionsStore> {
+
+    /** The set of reaching definitions in this store */
+    private final Set<ReachingDefinitionsNode> reachingDefSet;
+
+    /** Create a new ReachDefinitionStore. */
+    public ReachingDefinitionsStore() {
+        reachingDefSet = new LinkedHashSet<>();
+    }
+
+    /**
+     * Create a new ReachDefinitionStore.
+     *
+     * @param reachingDefSet a set of reaching definitions nodes, this parameter is captured and
+     *     that the caller should not retain an alias
+     */
+    public ReachingDefinitionsStore(LinkedHashSet<ReachingDefinitionsNode> reachingDefSet) {
+        this.reachingDefSet = reachingDefSet;
+    }
+
+    /**
+     * Remove the information of a reaching definition from the reaching definitions set.
+     *
+     * @param defTarget target of a reaching definition
+     */
+    public void killDef(Node defTarget) {
+        Iterator<ReachingDefinitionsNode> it = reachingDefSet.iterator();
+        while (it.hasNext()) {
+            ReachingDefinitionsNode generatedDefNode = it.next();
+            // It's preferred to use "==" to compare two nodes in checker framework,
+            // but in this case we use `equals` to only measure value equality.
+            // If we use "==", two expressions from different nodes with same
```

Comment:
Can you try to reword? Something reads odd here.

Reply:
Thanks!